3rd_party/CMakeLists.txt

```diff
@@ -0,0 +1 @@
+add_subdirectory(catch2)
```

Comment:
Can we spell 3rd_party as third_party to match e.g. grpc?

Reply:
I updated to use gtest and removed the 3rd_party directory

Reply:
Thanks!